test/extended/images/mysql_replica.go

```diff
@@ -0,0 +1,266 @@
+	PodName string
+	Env     map[string]string
+}
+
+func NewMysql(oc *exutil.CLI, podName, service string) (*mysql, error) {
+	mysql := &mysql{
+		oc:      oc,
+		PodName: podName,
+	}
+
+	pod, err := oc.KubeREST().Pods(oc.Namespace()).Get(podName)
+	if err != nil {
+		return nil, err
+	}
+	mysql.Env = make(map[string]string, len(pod.Spec.Containers[0].Env))
+	for _, env := range pod.Spec.Containers[0].Env {
+		mysql.Env[env.Name] = env.Value
+	}
+	return mysql, nil
+}
+
+// WaitForMySQL pings the MySQL server
+func (m *mysql) WaitForMySQL() (bool, error) {
+	out, err := m.oc.ExecWithoutExit(m.PodName, "mysqladmin -h "+"127.0.0.1"+" -u"+"root"+" ping").Output()
+	if err != nil {
+		switch err.(type) {
+		case *exutil.ExitError:
+			return false, nil
+		default:
+			return false, err
+		}
+	}
+	return out == "mysqld is alive", nil
+}
+
+// WaitUntilUp continuously waits for the server to respond to pings, up until timeout.
+func (m *mysql) WaitUntilUp(timeout time.Duration) error {
+	return wait.Poll(2*time.Second, timeout, m.WaitForMySQL)
+}
+
+// TODO: Generalize, move to exutil
+func (m *mysql) QueryAsRoot(sql string) (string, error) {
+	return m.oc.ExecWithoutExit(m.PodName, "mysql -h "+"127.0.0.1"+" -uroot -e \""+sql+"\" "+m.Env["MYSQL_DATABASE"]).Verbose().Output()
+}
+
+// TODO: Generalize, move to exutil
+func (m *mysql) QueryAsUser(sql string) (string, error) {
+	return m.oc.ExecWithoutExit(m.PodName, "mysql -h "+"127.0.0.1"+" -u"+m.Env["MYSQL_USER"]+" -p"+m.Env["MYSQL_PASSWORD"]+" -e \""+sql+"\" "+m.Env["MYSQL_DATABASE"]).Verbose().Output()
+}
+
+// TODO: Generalize, move to exutil
+func (m *mysql) WaitForMySQLOutput(timeout time.Duration, root bool, sql, resultSubstr string) error {
+	return wait.Poll(5*time.Second, timeout, func() (bool, error) {
+		out, err := m.QueryAsRoot(sql)
+		if err != nil {
+			return false, err
+		}
+		if strings.Contains(out, resultSubstr) {
+			return true, nil
+		}
+		return false, nil
+	})
+}
+
+var _ = g.Describe("images: MySQL replication template", func() {
+	defer g.GinkgoRecover()
+	var (
+		volumeTmpDir string
+		templatePath = "https://raw.githubusercontent.com/openshift/mysql/master/5.5/examples/replica/mysql_replica.json"
+		oc           = exutil.NewCLI("mysql-replication", exutil.KubeConfigPath())
+		user         = "user"
+		userPass     = "userpass"
+		rootPass     = "rootpass"
+		database     = "userdb"
+	)
+
+	g.Describe("MySQL replication", func() {
+		g.JustBeforeEach(func() {
+			var err error
+			volumeTmpDir, err = ioutil.TempDir(exutil.TestContext.OutputDir, "persistent-volume")
+			o.Expect(err).NotTo(o.HaveOccurred())
+			err = os.Chmod(volumeTmpDir, 0777)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			pv := exutil.CreatePersistentVolume("mysql-pv", "1Gi", volumeTmpDir)
+			_, err = oc.AdminKubeREST().PersistentVolumes().Create(pv)
+			o.Expect(err).NotTo(o.HaveOccurred())
+		})
+
+		g.AfterEach(func() {
+			err := oc.AsAdmin().Run("delete").Args("pods", "--all").Verbose().Execute()
```

Comment:
You can do this in one call to `oc` (and using the namespace is a good idea, can avoid hard to debug problems later):

```go
err = oc.AsAdmin().Run("delete").Args("all,pvc,pv", "--all", "-n", namespace).Verbose().Execute()
```